paasta_tools/mesos_maintenance.py

```diff
@@ -342,12 +513,13 @@ def drain(hostnames, start, duration):
     :returns: None
     """
     log.info("Draining: %s" % hostnames)
+    reserve_all_resources(hostnames)
     payload = build_maintenance_schedule_payload(hostnames, start, duration, drain=True)
     client_fn = get_schedule_client()
     try:
         drain_output = client_fn(method="POST", endpoint="", data=json.dumps(payload)).text
-    except HTTPError as e:
-        e.msg = "Error performing maintenance drain. Got error: %s" % e.msg
+    except HTTPError:
+        log.debug("Error performing maintenance drain.")
```

Comment:
Most (if not all) of these blocks were resulting in their own exceptions being thrown due to `e.msg` not existing. I could probably throw a new `HTTPError` with a more descriptive message rather than simply calling `raise`.